pkg/server/ca/ca.go

```diff
@@ -78,19 +78,26 @@ func (ca *serverCA) SignX509SVID(ctx context.Context, csrDER []byte, ttl time.Du
 	}
 	notBefore := now.Add(-backdate)
 	notAfter := now.Add(ttl)
-	if notAfter.After(kp.x509CA.NotAfter) {
-		notAfter = kp.x509CA.NotAfter
+	if notAfter.After(kp.x509CA.cert.NotAfter) {
+		notAfter = kp.x509CA.cert.NotAfter
 	}
 
 	serialNumber := big.NewInt(atomic.AddInt64(&ca.x509sn, 1))
 
 	template, err := CreateX509SVIDTemplate(csrDER, ca.c.TrustDomain.Host, notBefore, notAfter, serialNumber)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	km := ca.c.Catalog.KeyManagers()[0]
-	return x509util.CreateCertificate(ctx, km, template, kp.x509CA, kp.X509CAKeyId(), template.PublicKey)
+	cert, err := x509util.CreateCertificate(ctx, km, template, kp.x509CA.cert, kp.X509CAKeyID(), template.PublicKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// the chain contains every cert from the ca to the root that signed the
+	// ca. return all but the root as the list of intermediates.
+	return cert, kp.x509CA.chain[:len(kp.x509CA.chain)-1], nil
```

Comment:
no, the chain contains every cert in the chain. it will always be at least length two (one for the CA cert, one for the signing entity.... see the comment)